### ISSUE TITLE
Coordinate not found error for atmospheric data

### ISSUE THREAD

**Issue by hahsan1** [ESMValGroup/ESMValCore#1196]
I am trying to extract level data from a model file. This file uses the alternate hybrid sigma pressure coordinate.

I've been getting the error: `CoordinateNotFoundError: 'Expected to find exactly 1  coordinate, but found none.'`

The recipe I am using does work for other model files though, so I'm not sure what's wrong with this particular file. Perhaps something in the metadata?

I have attached the ncdump output for the file, as well the recipe I used and the log file. 

Thank you

[main_log.txt](https://github.com/ESMValGroup/ESMValTool/files/6719096/main_log.txt)
[main_log_debug.txt](https://github.com/ESMValGroup/ESMValTool/files/6719097/main_log_debug.txt)
[ncdump.txt](https://github.com/ESMValGroup/ESMValTool/files/6719098/ncdump.txt)
[my_recipe.txt](https://github.com/ESMValGroup/ESMValTool/files/6719103/my_recipe.txt)

**Comment by hahsan1** [comment]
Thank you @schlunma, I managed to get this working now!

First I used an nco command to change the formula field to `p(n,k,j,i) = ap(k) + b(k)*ps(n,j,i)` which is the appropriate formula for pressure. The formula it had before is for calculating level k (based on this  [CF conventions website](https://cfconventions.org/Data/cf-conventions/cf-conventions-1.0/build/apd.html)). 

Then I added the appropriate pressure units to ap and ap_bnds. So I didn't end up using the fix files, but your comments pointed me in the right direction.
